### PR TITLE
feat: add rate limiter for the ssh requests 

### PR DESCRIPTION
**Pull request by 4m1z**
add rate limiter for the ssh requests

### CODE REVIEW

File: RateLimiter.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package main
```
Comment (Gigaxel) [inline RateLimiter.go]: don't name files in pascal case; snake case is preferred

File: RateLimiter.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"time"
+
+	"github.com/gliderlabs/ssh"
+	"golang.org/x/time/rate"
```
Comment (Gigaxel) [inline RateLimiter.go]: I don't want us to introduce this 3rd party dependency, when we have everything we need to implement the rate limiter on our own with a few more lines of code, and leveraging redis, which we already have

**Review by Gigaxel — CHANGES_REQUESTED** [review]
The idea is nice, but we there is no need for a 3rd party library. Make sure to use redis, and implement your own custom middleware

File: RateLimiter.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+func RateLimiter(numberOfRequests int, handler ssh.Handler) ssh.Handler {
```
Comment (Gigaxel) [inline RateLimiter.go]: also just based on the first look, but it seems like you've rate limited every user? I don't see any way you identify a specific user that we want to rate limit.

**Review by Gigaxel — CHANGES_REQUESTED** [review]

File: RateLimiter.go (hunk 2 above)
Comment (4m1z) [reply RateLimiter.go]: agree with you, I will work on it

File: RateLimiter.go (hunk 3 above)
Comment (4m1z) [reply RateLimiter.go]: yes, this limiter is for every user, so I will make the limiter work per user.

File: RateLimiter.go (hunk 1 above)
Comment (4m1z) [reply RateLimiter.go]: renamed the file

**Comment by Gigaxel** [comment]
a new feature has been merged and deployed, so you'll need to get the latest changes on your branch first

**Comment by Gigaxel** [comment]
I'll have to close this PR, due to it being already implemented!

**Pull request closed without merging**